fluent-plugin-enhance-k8s-metadata/lib/fluent/plugin/filter_enhance_k8s_metadata.rb

```diff
@@ -83,7 +83,7 @@ def decorate_record(record)
             record.delete('service')
           end
           metadata = get_pod_metadata(namespace_name, pod_name)
-          service = @pods_to_services[pod_name]
+          service = @pods_to_services[pod_name] unless @pods_to_services.nil?
```

Comment:
does it mean we are going to not tag service properly until some background job is done?

Reply:
Yes I believe this is how it works (and used to work before as well, only with an error which is now prevented).

Reply:
Do other lines (below this one) make sense when `service` is not assigned here (it would be nil then, right?)

Reply:
The next line checks if `service` is `nil`, so it seems OK to me:
https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/1b6ce8eed201269b7a29f92b2218bbc2bcba3817/fluent-plugin-enhance-k8s-metadata/lib/fluent/plugin/filter_enhance_k8s_metadata.rb#L87

Reply:
> Yes I believe this is how it works (and used to work before as well, only with an error which is now prevented).

Not sure, error prevented from ingesting data if `@pods_to_services` was not initialised, now we do not wait for initialisation and part of the records won't get metadata even if there is service for pod :confused:

Reply:
We did some investigation and it occurs that when a Fluentd plugin crashes on a record, Fluentd does not drop o retry the record which caused the crash, it just happily passes the record (possibly half-modified by the plugin) further into the processing pipeline.